torch/csrc/serialization.cpp

```diff
@@ -27,6 +32,15 @@ static Py_ssize_t doPartialPythonWrite(
 
 template <>
 Py_ssize_t doPartialRead<int>(int fildes, void* buf, size_t nbytes) {
+  opentelemetry::trace::StartSpanOptions start;
+  start.start_system_time = SystemTimestamp(std::chrono::nanoseconds(300));
+  std::shared_ptr<opentelemetry::metrics::Counter<uint64_t>> counter{
+          new opentelemetry::metrics::NoopCounter<uint64_t>("testi2", "none", "unitless")};
```

Comment:
rather than binding to concrete implementation you want to pull it from the Meter:
```
auto meter = opentelemetry::metrics::Provider::GetMeterProvider()->GetMeter(...);
auto counter = meter->CreateUInt64Counter(...);
```
Also, it's more efficient to create counter once as static or member variable.